bazel-mode.el

```diff
@@ -21,6 +21,43 @@
 ;;
 ;;; Code:
 
+(defgroup bazel-mode nil
+  "Major mode for Bazel BUILD files."
+  :link '(url-link "https://github.com/bazelbuild/emacs-bazel-mode")
+  :group 'languages)
+
+(defcustom bazel-mode-buildifier-cmd "buildifier"
+  "Filename of buildifier executable."
+  :type 'file
+  :group 'bazel-mode
+  :link '(url-link
+          "https://github.com/bazelbuild/buildtools/tree/master/buildifier"))
+
+(defun bazel-mode-buildifier ()
+  "Format current buffer using buildifier."
+  (interactive "*")
+  (let ((input-buffer (current-buffer))
+        (buildifier-buffer (get-buffer-create "*buildifier*"))
+        ;; Run buildifier on a file to support remote BUILD files.
+        (temp-file (make-nearby-temp-file "buildifier")))
+    (write-region nil nil temp-file nil 1)
+    (with-current-buffer buildifier-buffer
+      (setq-local inhibit-read-only t)
+      (erase-buffer)
+      (unwind-protect
+        (let ((return-code
+               (process-file bazel-mode-buildifier-cmd temp-file
+                             buildifier-buffer nil "-type=build")))
```

Comment:
Passing a single buffer mixes standard output and error, which can cause incorrect behavior if buildifier prints log messages to stderr. You should probably use a separate file for standard error by passing a list to *buffer*.

Reply:
Done. I'm reusing the temp file created to hold the input. I could separate this use of the temp file into one specifically for input and a distinct file specifically for holding stderr, if you think that helps.

Reply:
Pretty sure this won't work as the input file will then be truncated before processing even starts. I guess there's no way around using two temporary files.

Reply:
It worked for me in local tests, but let's err on the side of safety and just split the two purposes into two distinct files.

Reply:
I've done this already, btw.